src/tree_store/btree.rs

```diff
@@ -726,6 +738,71 @@ impl<K: Key, V: Value> Btree<K, V> {
         }
     }
 
+    pub(crate) fn first(
+        &self,
+    ) -> Result<Option<(AccessGuard<'static, K>, AccessGuard<'static, V>)>> {
+        if let Some(ref root) = self.cached_root {
+            self.first_helper(root.clone())
+        } else {
+            Ok(None)
+        }
+    }
+
+    fn first_helper(
+        &self,
+        page: PageImpl,
+    ) -> Result<Option<(AccessGuard<'static, K>, AccessGuard<'static, V>)>> {
+        let node_mem = page.memory();
+        match node_mem[0] {
+            LEAF => {
+                let accessor = LeafAccessor::new(page.memory(), K::fixed_width(), V::fixed_width());
+                let (key_range, value_range) = accessor.entry_ranges(0).unwrap();
+                let key_guard = AccessGuard::with_page(page.clone(), key_range);
+                let value_guard = AccessGuard::with_page(page, value_range);
+                Ok(Some((key_guard, value_guard)))
+            }
+            BRANCH => {
+                let accessor = BranchAccessor::new(&page, K::fixed_width());
+                let child_page = accessor.child_page(0).unwrap();
+                self.first_helper(self.mem.get_page_extended(child_page, self.hint)?)
```

Comment:
I understand tree depth is somewhat limited, but this looks highly amenable to an iterative implementation by basically putting all of this in a loop and writing

```rust
page = self.mem.get_page_extended(child_page, self.hint)?;
```

here with a

```
return Ok(Some((key_guard, value_guard)));
```

above to avoid the any questions of call overhead and/or stack usage.